pkg/detectors/braintreepayments/braintreepayments.go

```diff
@@ -0,0 +1,99 @@
+package braintreepayments
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	//Make sure that your group is surrounded in boundry characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"braintreepayments"}) + `\b([0-9a-f]{32})\b`)
+	idPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"braintreepayments"}) + `\b([0-9a-z]{16})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"braintreepayments"}
+}
+
+// FromData will find and optionally verify BraintreePayments secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	idMatches := idPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+
+		for _, idMatch := range idMatches {
+			if len(idMatch) != 2 {
+				continue
+			}
+
+			resIdMatch := strings.TrimSpace(idMatch[1])
+
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_BraintreePayments,
+				Raw:          []byte(resMatch),
+			}
+
+			if verify {
+				payload := strings.NewReader(`{"query": "query { ping }"}`)
+				req, err := http.NewRequestWithContext(ctx, "POST", "https://payments.sandbox.braintree-api.com/graphql", payload)
+				if err != nil {
+					continue
+				}
+				req.Header.Add("Content-Type", "application/json")
+				req.Header.Add("Braintree-Version", "2019-01-01")
+				req.SetBasicAuth(resIdMatch, resMatch)
+				res, err := client.Do(req)
+				if err == nil {
+					bodyBytes, err := ioutil.ReadAll(res.Body)
+					if err != nil {
+						continue
+					}
+
+					bodyString := string(bodyBytes)
+					errorResponse := strings.Contains(bodyString, `"errorType":"developer_error"`)
+
+					defer res.Body.Close()
+					if res.StatusCode >= 200 && res.StatusCode < 300 {
```

Comment:
updated the validation. thanks!

Reply:
When it's possible, prefer to check for a substring in a valid response. Otherwise this can false positive under certain conditions (deploy, downtime, etc). We've encountered that before on some other detectors.

Reply:
Got it. Updated the validation to check for valid response is in the latest commit.